sdk/maps/maps-route-rest/README.md

```diff
@@ -0,0 +1,191 @@
+
+**Please rely heavily on our [REST client docs](https://github.com/Azure/azure-sdk-for-js/blob/main/documentation/rest-clients.md) to use this library**
+
+Key links:
+
+- [Source code][source_code]
+- [Package (NPM)][npm_link]
+- [API reference documentation][api_ref]
+- [Product Information](https://docs.microsoft.com/rest/api/maps/route)
+
+## Getting started
+
+### Currently supported environments
+
+- [LTS versions of Node.js](https://nodejs.org/about/releases/)
+- Latest versions of Safari, Chrome, Edge and Firefox.
+
+### Prerequisites
+
+- An [Azure subscription][azure_sub].
+- An [Azure Maps account](https://docs.microsoft.com/azure/azure-maps/how-to-manage-account-keys). You can create the resource via the [Azure Portal][azure_portal], the [Azure PowerShell][azure_powershell], or the [Azure CLI][azure_cli].
+
+If you use Azure CLI, replace `<resource-group-name>` and `<map-account-name>` of your choice, and select a proper [pricing tier](https://docs.microsoft.com/azure/azure-maps/choose-pricing-tier) based on your needs via the `<sku-name>` parameter. Please refer to [this page](https://docs.microsoft.com/cli/azure/maps/account?view=azure-cli-latest#az_maps_account_create) for more details.
+
+### Install the `@azure-rest/maps-route` package
+
+Install the Azure MapsRoute client REST client library for JavaScript with `npm`:
+
+```bash
+npm install @azure-rest/maps-route
+```
+
+### Create and authenticate a `MapsRouteClient`
+
+To create a client object to access the Azure Maps Route APIs, you will need a `credential` object. The Azure Maps Route client can use an Azure Active Directory credential or an Azure Key credential to authenticate.
+
+#### Using an Azure Active Directory Credential
+
+You can authenticate with Azure Active Directory using the [Azure Identity library][azure_identity]. To use the [DefaultAzureCredential][defaultazurecredential] provider shown below, or other credential providers provided with the Azure SDK, please install the `@azure/identity` package:
+
+```bash
+npm install @azure/identity
+```
+
+You will also need to register a new AAD application and grant access to Azure Maps by assigning the suitable role to your service principal. Please refer to the [Manage authentication](https://docs.microsoft.com/azure/azure-maps/how-to-manage-authentication) page.
+
+Set the values of the client ID, tenant ID, and client secret of the AAD application as environment variables: `AZURE_CLIENT_ID`, `AZURE_TENANT_ID`, `AZURE_CLIENT_SECRET`.
+
+You will also need to specify the Azure Maps resource you intend to use by specifying the `clientId` in the client options.
+The Azure Maps resource client id can be found in the Authentication sections in the Azure Maps resource. Please refer to the [documentation](https://docs.microsoft.com/azure/azure-maps/how-to-manage-authentication#view-authentication-details) on how to find it.
+
+```javascript
+const { createMapsRouteClient } = require("@azure-rest/maps-route");
```

Comment:
I think here we need:

```js
const MapsRouteClient = require("@azure-rest/maps-route").default;
```

And later on

```js
const client = MapsRouteClient(credential, "<maps-account-client-id>");
```

Reply:
Thanks for pointing this out! There are two topics I would like to discuss.

First, I use named export for `createMapsRouteClient` to follow the eslint rules [ts-module-only-named](https://github.com/Azure/azure-sdk-for-js/blob/main/common/tools/eslint-plugin-azure-sdk/docs/rules/ts-modules-only-named.md). But not sure if this need to be applied to RLC.

Second, is the pattern we need to follow that the client's factory function needs to start with a capital character (`Mapsxxxx`) across all the SDKs? I think `MapsRouteClient` sounds more like a class to me, so I add the prefix `create-`.

Reply:
According to the previous conversation, we want to emphasize that `MapsRoute` is the only function we need for RLC. That's the reason we prefer default export here. Changed in eecbf537cc707c8b2df9c8cd09ded8837972c951